app/components/Settings/EncryptForm/EncryptForm.jsx

```diff
@@ -96,14 +94,67 @@ export default class EncryptForm extends React.Component<Props, State> {
     )
   }
 
+  validatePrivateKey = (privateKey: string) => {
```

Comment:
chose to validate here (and only throw errors the action) so that error messages can be targeted to each input.
eg. invalid private key - error shown on private key field